### PR TITLE
Hsr05 patch

### PR DESCRIPTION
**Pull request by HSR05**
- [x] Tests added / passed
- [x] Passes `flake8 dask`

### CODE REVIEW

File: dask/dataframe/io/demo.py [hunk 1]
```diff
@@ -41,7 +41,11 @@ def make_categorical(n, rstate):
         'category': make_categorical}
 
 
+hsr05-patch
```
Comment (TomAugspurger) [inline dask/dataframe/io/demo.py]: Bad merge conflict resolution here?

File: dask/dataframe/utils.py [hunk 2]
```diff
@@ -379,15 +379,15 @@ def _nonempty_index(idx):
         return typ([1, 2], name=idx.name)
     elif typ is pd.Index:
         return pd.Index(['a', 'b'], name=idx.name)
-    elif typ is pd.DatetimeIndex:
+    elif typ is pd.date_range:
```
Comment (TomAugspurger) [inline dask/dataframe/utils.py]: This won't work. `pd.date_range` is a function. Leaving DatetimeIndex should be fine.

File: dask/dataframe/io/demo.py (hunk 1 above)
Comment (HSR05) [reply dask/dataframe/io/demo.py]: That was my mistake. Sorry.

File: dask/dataframe/utils.py (hunk 2 above)
Comment (HSR05) [reply dask/dataframe/utils.py]: resolved.

Comment (HSR05) [reply dask/dataframe/utils.py]: Is it okay now?

File: dask/dataframe/io/io.py [hunk 3]
```diff
@@ -101,7 +101,7 @@ def from_array(x, chunksize=50000, columns=None):
     return new_dd_object(dsk, name, meta, divisions)
 
 
-def from_pandas(data, npartitions=None, chunksize=None, sort=True, name=None):
+def from_pandas(data, npartitions=10, chunksize=None, sort=True, name=None):
```
Comment (jrbourbeau) [inline dask/dataframe/io/io.py]: I'm not sure why the default value of `npartitions` was changed here. Could you elaborate on your reasoning for this @HSR05?

**Comment by mrocklin** [comment]
@HSR05 there seems to be a number of different things going on in this PR.  Ideally a PR should close only one issue.  It's somewhat difficult to review otherwise.

**Comment by HSR05** [comment]
> 
> 
> It looks like you still have changes from other issues here. What issue is this supposed to close?

Closes #4568 

I'll make a PR for other issues by making another branch to make it easier to review. Sorry for all the inconvenience.

**Comment by HSR05** [comment]
@mrocklin i made a mistake of not making another branch for a new issue. What should i do now as this PR was closing #4568 . Can you please review it as i am also going to apply for GSoC. So i can resolve the problems and send PR for other issues that i mixed in this PR.

**Comment by mrocklin** [comment]
This seems to have gone stale.  Closing.

**Pull request closed without merging**